internal/services/dataprotection/data_protection_backup_vault_resource.go

```diff
@@ -229,16 +240,24 @@ func resourceDataProtectionBackupVaultRead(d *pluginsdk.ResourceData, meta inter
 	if model := resp.Model; model != nil {
 		d.Set("location", location.NormalizeNilable(model.Location))
 		props := model.Properties
-		if len(props.StorageSettings) > 0 {
+
+		if props.StorageSettings != nil && len(props.StorageSettings) > 0 {
 			d.Set("datastore_type", string(pointer.From((props.StorageSettings)[0].DatastoreType)))
 			d.Set("redundancy", string(pointer.From((props.StorageSettings)[0].Type)))
 		}
+
 		if securitySetting := model.Properties.SecuritySettings; securitySetting != nil {
+			if immutability := securitySetting.ImmutabilitySettings; immutability != nil {
+				if immutability.State != nil {
```

Comment:
The way this is currently written is a breaking change. Moving from the current provider version to this change causes the following diff

```
      + immutability                 = "0"
```

We should set it to the default regardless of whether immutability state is nil